### ISSUE TITLE
Z-waif can't reach oobabooga

### ISSUE THREAD

**Issue by Tonttis** [SugarcaneDefender/z-waif#2]
Otherwise everything on setup is ok, but can't get z-waif connect to oobabooga.

Followed your setup video and everything works fine except connection between z-waif and oobabooga.

If you have a idea what causes this, please let me know. :)

Edit: I have toggled the api on, --api command.

**Comment by SugarcaneDefender** [comment]
I think it might have to do with the API port. How old is your install of Oobabooga? I know they changed the whole format API like 7+ months ago, so it could be failing to communicate with it. If it is older, you could install the OpenAI API module instead of the normal one, if you don't/can't update it.

It could also be a newer install of it having issues; they did update oobabooga recently, and I read there was some API changes in the updates (which, would be really annoying for me lol). How old is your install? Also, when you first launch Oooba, it should pop up in the console log saying "API open on port XXXX", what port is it opening on?

**Comment by Tonttis** [comment]
I installed Oobabooga 10th of October, so not old but relatively new. 

Port 5000 is being opened for the API.

**Comment by SugarcaneDefender** [comment]
Ah, I see. Can you send your log.txt file, located in the main z-waif folder? And could you also send the contents of your .env file (opened in notepad)? One of the few things that could be causing the issues I can think of:

- Model not being loaded in Oobabooga (or it's broken in it; I always test in the text chat within Oooba itself when I set up)
- Mismatched character name in the .env to a character in Oobabooga (special characters and spaces may mess this up???)
- Issues with generation rules

If the oobabooga console window keeps showing new "Output Generated" messages, then it is likely the last one. If I can't figure it out, I can always make and then send a quick debug script to you and we can break it down further, and see where exactly it is going wrong!

**Comment by Tonttis** [comment]
[log.txt](https://github.com/user-attachments/files/17490481/log.txt)
![image](https://github.com/user-attachments/assets/53f3b372-b14d-4696-ba46-b27685648b8c)

Heres the log.txt and .env in notepad. It seems that log.txt for sure has some errors, I'll try to troubleshoot them too.

- Model loads fine, as messaging works.
- The name is the same, and generic.
- Not 100% sure what "generation rules" means, but I've followed your tutorial for those, I think. :)

Thanks for the help!

**Comment by Tonttis** [comment]
So, just noticed that building the wheel in installation fails.

Getting requirements to build wheel ... error

I had not seen this before as it is just in plain white as everything else.

**Comment by SugarcaneDefender** [comment]
Alrighty, I just released a new update. This should fix the UI (Gradio forgot to check if their dependencies actually work or not lol) - And maybe your Ooobabooga issues? Did you have no mic connected before, and just receive blank messages? There was a bug where, when it was initializing, it would crash if none was detected. Should be fixed now.

I would download Z-Waif again and install in another folder.

I also noticed, while troubleshooting, that I misconfigured my name - and also was using the latest Ooba install, and it would give me error messages. So the latest Ooba is confirmed working, you have a Character Card set up in Ooba right? If you get no errors logged in the Ooba console, then it's likely something in my program still.

Let me know if it works!
